### DISCUSSION TITLE
Histogram Panel

### DISCUSSION THREAD

**Discussion by valschmidt** [foxglove/community#9] [Q&A]
Hello Foxglove!

Recently we found ourselves troubleshooting (reverse engineering really) an algorithm involved in automatically docking a robotic boat. The algorithm relies heavily on lidar point cloud data, which is pre-filtered by return intensity. We realized it would be really helpful to have a new Foxglove Studio Histogram panel to better gauge the range of sensor data inputs. Here are the features as we envision it:

- Plots a bar-type histogram of any field over a fixed window in time - defaults to some reasonable value, but generally user-specified (which may be infinite)
- Auto-generates the x-axis bin sizes for 10 bins by default, but optionally allows them to be user specified (by min, max, number of bins) 
- Optionally pre-filters data, omitting data below or above user-specified bounds. Allows one to omit outliers. 
- Plots multiple histograms on a single plot with a common set of X-axis bins, with different colors for each data source. Overlapping histograms might have some transparency to underlying ones are visible?
- Optionally provide a text indication of the mean and variance of the data.
- When the input data is not numeric, the histogram panel might produce a bar graph counting the categories of each identical field received.

**Comment by defunctzombie** [comment]
Moving this over to discussion for more input/brainstorming. Sounds like a good idea for a custom extension.

**Comment by ilikeyu** [comment]
is there an existing histogram panel extension now?

**Reply by amacneil** [reply]
Not that I'm aware of, but it would be a great extension panel.
